File: AltonikaShop.Domain/Entities/Product.cs
using System.Collections.Generic;
using CoreLib.Entities;

namespace AltonikaShop.Domain.Entities
{
    public class Product : NamedEntity
    {
        public string Code { get; set; }

        public decimal? Price { get; set; }

        public string Description { get; set; }
        
        public string ImageSource { get; set; }

        public bool IsDeleted { get; set; }

        public static IList<string> Tvp = new List<string>
        {
            nameof(Id),
            nameof(Name)
        };
    }
}

File: 2.0/AltonikaShop/AltonikaShop.Web/ClientApp/app/entities/product.ts
import { NamedEntity } from 'nglib';
import { BasketItem } from './basket-item';

export class Product extends NamedEntity {
    
    code: string;
    price: number;
    description: string;
    imageSource: string;
    isDeleted: boolean;
    isFavorite: boolean;
    
    constructor() {
        super();
    }
}
